src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -911,20 +933,27 @@ public VariantContextWriter createVCFWriter(final Path outPath) {
         if (outputSitesOnlyVCFs) {
             options.add(Options.DO_NOT_WRITE_GENOTYPES);
         }
-
+        final VariantContextWriter unfilteredWriter;
         if (maxVariantsPerShard > 0) {
-            return new ShardingVCFWriter(
+            unfilteredWriter = new ShardingVCFWriter(
                     outPath,
                     maxVariantsPerShard,
                     sequenceDictionary,
                     createOutputVariantMD5,
-                    options.toArray(new Options[options.size()]));
+                    options.toArray(new Options[0]));
+        } else {
+            unfilteredWriter = GATKVariantContextUtils.createVCFWriter(
```

Comment:
While this hook seems reasonable, i think perhaps its a little cleaner and more extensible to construct this thing as an option within createVCFWriter which on casual inspection seems to be used by a lot of tools (at least haplotypeCaller + Mutect). Given that this IS a GATKTool argument i'm a little worried that its not going to work for a reasonable chunk of our tools.

Reply:
the writers do not have access to the intervals, thus its best to just put this in GATKtool which does rather than threading it all the way down to these things....